lib/components/user/trip-summary.js

```diff
@@ -1,74 +1,51 @@
 import coreUtils from '@opentripplanner/core-utils'
 import { ClassicLegIcon } from '@opentripplanner/icons'
-import React, { Component } from 'react'
+import React, { useEffect, useState } from 'react'
 import { connect } from 'react-redux'
 
 import { getQueryParamsFromQueryString } from '../../util/query'
 import ItinerarySummary from '../narrative/default/itinerary-summary'
 
 const { formatDuration, formatTime } = coreUtils.time
 
-class TripSummary extends Component {
-  constructor () {
-    super()
-
-    this.state = {
-      // Hold a default state for query location names
-      // while it is being fetched in updateQueryState.
-      queryObj: {
-        from: {},
-        to: {}
-      }
-    }
-  }
-
-  updateQueryState = async () => {
-    const { monitoredTrip, config } = this.props
-    const { queryParams } = monitoredTrip
-    const queryObj = await getQueryParamsFromQueryString(queryParams, config)
-
-    // Prevent a state update if componentWillUnmount happened while await was executing.
-    if (!this.state.willUnmount) {
-      this.setState({ queryObj })
-    }
-  }
-
-  componentDidMount () {
-    this.updateQueryState()
-  }
-
-  componentDidUpdate (prevProps) {
-    // Update the state the monitoredTrip prop has changed.
-    if (this.props.monitoredTrip !== prevProps.monitoredTrip) {
-      this.updateQueryState()
+const TripSummary = ({ config, monitoredTrip }) => {
```

Comment:
These changes really seem like things are headed in the wrong direction... all of this state management stuff shouldn't be necessary... maybe we should just go over this together to make sure things are working right. I don't think we're really improving anything by changing from a class to functional component -- just reformatting the weird behavior.

Reply:
Got rid of that overhead per your advice @landonreed in https://github.com/opentripplanner/otp-react-redux/pull/174/commits/0281f40e7a248fcbcacc4a553176d550baaef995.